### PR TITLE
support read git commits history

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/docfx/config/Config.cs [hunk 1]
```diff
@@ -184,6 +184,11 @@ private static void ValidateContributorConfig(ContributionConfig config, string
             {
                 throw Errors.UserProfileCacheNotFound(config.UserProfileCachePath).ToException();
             }
+            if (!string.IsNullOrEmpty(config.GitCommitsHistoryPath)
+                && File.Exists(Path.Combine(configPath, config.GitCommitsHistoryPath)))
+            {
+                throw Errors.UserProfileCacheNotFound(config.UserProfileCachePath).ToException();
```
Comment (yufeih) [inline src/docfx/config/Config.cs]: GitCommitsHistoryPath

File: src/docfx/lib/git/GitCommitsHistory.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    internal class GitCommitsHistory
+    {
+        [JsonProperty("last_build_commit_id")]
+        public string LastBuildCommitId { get; set; }
+
+        [JsonProperty("commits")]
+        public List<CommitsHistoryItem> Commits { get; set; } = new List<CommitsHistoryItem>();
+
+        [JsonProperty("cross_repository_references")]
+        public Dictionary<string, GitCommitsHistory> CrossRepositoryReferences { get; set; }
```
Comment (yufeih) [inline src/docfx/lib/git/GitCommitsHistory.cs]: remove LastBuildCommitId / CrossRepositoryReferences

File: src/docfx/config/ContributionConfig.cs [hunk 3]
```diff
@@ -26,5 +26,10 @@ internal class ContributionConfig
         /// The path of user profile cache, used for generating authoer and contributors.
         /// </summary>
         public readonly string UserProfileCachePath = string.Empty;
+
+        /// <summary>
+        /// The path of commit time history file, which contains the time each commit being pushed.
+        /// </summary>
+        public readonly string GitCommitsHistoryPath = string.Empty;
```
Comment (yufeih) [inline src/docfx/config/ContributionConfig.cs]: nit: name it `GitCommitTime` because it does not have history.

File: docs/specs/git.yml [hunk 4]
```diff
@@ -27,4 +27,9 @@ outputs:
       "enableContribution": true,
       "editLink": "https://github.com/docascode/docfx-test-dependencies/blob/git-test/README.md"
     }
+  testdata1.json: |
```
Comment (yufeih) [inline docs/specs/git.yml]: this breaks other builds

Comment (superyyrrzz) [reply docs/specs/git.yml]: open #2970 to fix it, because of login issue...

**Pull request closed without merging**